yogstation/code/game/gamemodes/shadowling/special_shadowling_abilities.dm

```diff
@@ -0,0 +1,193 @@
+	name = "Hatch"
+	desc = "Casts off your disguise."
+	panel = "Shadowling Evolution"
+	charge_max = 3000
+	human_req = 1
+	clothes_req = 0
+	action_icon = 'yogstation/icons/mob/actions.dmi'
+	action_icon_state = "hatch"
+
+/obj/structure/alien/resin/wall/shadowling //For chrysalis
+	name = "chrysalis wall"
+	desc = "Some sort of purple substance in an egglike shape. It pulses and throbs from within and seems impenetrable."
+	max_integrity = INFINITY
+
+/obj/effect/proc_holder/spell/self/shadowling_hatch/cast(list/targets,mob/user = usr)
+	if(user.stat || !ishuman(user) || !user || !is_shadow(user || isinspace(user)))
+		return
+	var/mob/living/carbon/human/H = user
+	var/hatch_or_no = alert(H,"Are you sure you want to hatch? You cannot undo this!",,"Yes","No")
+	switch(hatch_or_no)
+		if("No")
+			to_chat(H, "<span class='warning'>You decide against hatching for now.")
+			charge_counter = charge_max
+			return
+		if("Yes")
+			H.SetStun(INFINITY) //This is bad but hulks can't be stunned with the actual procs. I'm sorry.
+			H.visible_message("<span class='warning'>[H]'s things suddenly slip off. They hunch over and vomit up a copious amount of purple goo which begins to shape around them!</span>", \
+							"<span class='shadowling'>You remove any equipment which would hinder your hatching and begin regurgitating the resin which will protect you.</span>")
+
+			var/temp_flags = H.status_flags
+			H.status_flags |= GODMODE //Can't die while hatching
+
+			H.unequip_everything()
+
+			sleep(50)
+
+			var/turf/shadowturf = get_turf(user)
+			for(var/turf/open/floor/F in orange(1, user))
+				new /obj/structure/alien/resin/wall/shadowling(F)
+
+			for(var/obj/structure/alien/resin/wall/shadowling/R in shadowturf) //extremely hacky
+				qdel(R)
+				new /obj/structure/alien/weeds/node(shadowturf) //Dim lighting in the chrysalis -- removes itself afterwards
+
+			H.visible_message("<span class='warning'>A chrysalis forms around [H], sealing them inside.</span>", \
+							"<span class='shadowling'>You create your chrysalis and begin to contort within.</span>")
+
+			sleep(100)
+			H.visible_message("<span class='warning'><b>The skin on [H]'s back begins to split apart. Black spines slowly emerge from the divide.</b></span>", \
+							"<span class='shadowling'>Spines pierce your back. Your claws break apart your fingers. You feel excruciating pain as your true form begins its exit.</span>")
+
+			sleep(90)
+			H.visible_message("<span class='warning'><b>[H], skin shifting, begins tearing at the walls around them.</b></span>", \
+							"<span class='shadowling'>Your false skin slips away. You begin tearing at the fragile membrane protecting you.</span>")
+
+			sleep(80)
+			playsound(H.loc, 'sound/weapons/slash.ogg', 25, 1)
+			to_chat(H, "<i><b>You rip and slice.</b></i>")
+			sleep(10)
+			playsound(H.loc, 'sound/weapons/slashmiss.ogg', 25, 1)
+			to_chat(H, "<i><b>The chrysalis falls like water before you.</b></i>")
+			sleep(10)
```

Comment:
You need to add checks to see if `H`` is still valid/not qdelled after every sleep. Otherwise it runtimes.